src/e2e/java/teammates/e2e/cases/e2e/InstructorFeedbackSessionsPageE2ETest.java

```diff
@@ -0,0 +1,288 @@
+                InstructorFeedbackSessionsPage.class);
+        feedbackSessionsPage.waitForPageToLoad();
+        feedbackSessionsPage.verifySessionDetails(copiedSession2);
+        verifyPresentInDatastore(copiedSession2);
+
+        ______TS("publish results");
+        openSession.setResultsVisibleFromTime(Const.TIME_REPRESENTS_NOW);
+        feedbackSessionsPage.publishSessionResults(openSession);
+
+        feedbackSessionsPage.verifyStatusMessage("The feedback session has been published. "
+                + "Please allow up to 1 hour for all the notification emails to be sent out.");
+        feedbackSessionsPage.verifySessionDetails(openSession);
+        verifySessionPublishedState(openSession, true);
+        verifyEmailSent(studentToEmail.getEmail(), "TEAMMATES: Feedback session results published"
+                + " [Course: " + copiedCourse.getName() + "][Feedback Session: "
+                + openSession.getFeedbackSessionName() + "]");
+
+        ______TS("send reminder email");
+        feedbackSessionsPage.sendReminderEmail(openSession, studentToEmail);
+
+        feedbackSessionsPage.verifyStatusMessage("Reminder e-mails have been sent out to those students"
+                + " and instructors. Please allow up to 1 hour for all the notification emails to be sent out.");
+        verifyEmailSent(studentToEmail.getEmail(), "TEAMMATES: Feedback session reminder"
+                + " [Course: " + copiedCourse.getName() + "][Feedback Session: "
+                + openSession.getFeedbackSessionName() + "]");
+
+        ______TS("resend results link");
+        feedbackSessionsPage.resendResultsLink(openSession, studentToEmail);
+
+        feedbackSessionsPage.verifyStatusMessage("Session published notification emails have been resent"
+                + " to those students and instructors. Please allow up to 1 hour for all the notification emails to be"
+                + " sent out.");
+        verifyEmailSent(studentToEmail.getEmail(), "TEAMMATES: Feedback session results published"
+                + " [Course: " + copiedCourse.getName() + "][Feedback Session: "
+                + openSession.getFeedbackSessionName() + "]");
+
+        ______TS("unpublish results");
+        openSession.setResultsVisibleFromTime(Const.TIME_REPRESENTS_LATER);
+        feedbackSessionsPage.unpublishSessionResults(openSession);
+
+        feedbackSessionsPage.verifyStatusMessage("The feedback session has been unpublished.");
+        feedbackSessionsPage.verifySessionDetails(openSession);
+        verifySessionPublishedState(openSession, false);
+        verifyEmailSent(studentToEmail.getEmail(), "TEAMMATES: Feedback session results unpublished"
+                + " [Course: " + copiedCourse.getName() + "][Feedback Session: "
+                + openSession.getFeedbackSessionName() + "]");
+
+        ______TS("download results");
+        feedbackSessionsPage.downloadResults(openSession);
+        ThreadHelper.waitFor(5000);
```

Comment:
Maybe use retries here, where the success condition is the existence of downloaded file?

Reply:
Added it to the `verifyDownloadedFile` method